### ISSUE TITLE
[Bug]: PulsarContainer startup timeout

### ISSUE THREAD

**Issue by VMarisevs** [testcontainers/testcontainers-java#5830]
### Module

Pulsar

### Testcontainers version

1.17.3

### Using the latest Testcontainers version?

Yes

### Host OS

Mac

### Host Arch

Apple M1 Pro

### Docker version

```shell
docker version 4.11.1 (84025)
```

### What happened?

I have an integration test that should start pulsar and read a few messages from it.
It runs fine in jenkins on remote, but locally I cannot start the test as it fails with the timeout specified. It takes an 31s to start pulsar and it fails.
I tried to override Startup Check Strategy and Waiting strategy provided by `GenericContainer` with no luck.
Then I started to look into the `PulsarContainer` implementation why it is ignoring new configurations on line 114 it does **hard override any client configurations** and breaks the` GenericContainer` interface.

https://github.com/testcontainers/testcontainers-java/blob/master/modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java#L114

### Relevant log output

```shell
Container startup failed
org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.junit.platform.commons.util.ReflectionUtils.newInstance(ReflectionUtils.java:550)
	at org.junit.jupiter.engine.execution.ConstructorInvocation.proceed(ConstructorInvocation.java:56)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestClassConstructor(InvocationInterceptor.java:73)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:77)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestClassConstructor(ClassBasedTestDescriptor.java:355)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:302)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:79)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:280)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$4(ClassBasedTestDescriptor.java:272)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$5(ClassBasedTestDescriptor.java:271)
	at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:31)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$before$2(ClassBasedTestDescriptor.java:197)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:196)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:80)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
	... 56 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:542)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 57 more
Caused by: org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:70)
	at org.rnorth.ducttape.timeouts.Timeouts.doWithTimeout(Timeouts.java:60)
	at org.testcontainers.containers.wait.strategy.WaitAllStrategy.waitUntilReady(WaitAllStrategy.java:54)
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:953)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:485)
	... 59 more
Caused by: java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:65)
```

**Comment by VMarisevs** [comment]
PR that should solve the problem
https://github.com/testcontainers/testcontainers-java/pull/5674/files

**Comment by edjanmichiles-toast** [comment]
I'm facing the same issue after upgraded to the latest version (1.17.3)

**Comment by kiview** [comment]
Fixed by #5674, will be part of the next release.

**Comment by jxnu-liguobin** [comment]
@kiview Similar issues were found in version 1.17.5.  Could you inform us of potential issues? thank you
My env is  Mac M1, Docker has been started correctly
```
com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000009: connection released [route: {}->unix://localhost:2375][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
16:19:40.702 [ducttape-0] INFO org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /vibrant_galileo: Waiting for 30 seconds for URL: http://localhost:54079/admin/v2/clusters (where port 54079 maps to container port 8080)
16:20:10.708 [ZScheduler-2] DEBUG 🐳 [apachepulsar/pulsar:2.9.3] - Wait strategy threw an exception
org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:70)
	at org.rnorth.ducttape.timeouts.Timeouts.doWithTimeout(Timeouts.java:60)
	at org.testcontainers.containers.wait.strategy.WaitAllStrategy.waitUntilReady(WaitAllStrategy.java:54)
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:953)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:485)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)
	at com.dimafeng.testcontainers.SingleContainer.start(Container.scala:47)
	at star.testkit.PulsarLayer.pulsarContainer$$anonfun$1$$anonfun$1$$anonfun$1(PulsarLayer.scala:17)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at zio.ZIOCompanionVersionSpecific.attempt$$anonfun$1(ZIOCompanionVersionSpecific.scala:107)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:967)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:889)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:889)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:889)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:889)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:889)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:889)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:818)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:936)
	at zio.internal.FiberRuntime.evaluateEffect(FiberRuntime.scala:330)
	at zio.internal.FiberRuntime.evaluateMessageWhileSuspended(FiberRuntime.scala:448)
	at zio.internal.FiberRuntime.drainQueueOnCurrentThread(FiberRuntime.scala:223)
	at zio.internal.FiberRuntime.run(FiberRuntime.scala:140)
	at zio.internal.ZScheduler$$anon$3.run(ZScheduler.scala:424)
Caused by: java.util.concurrent.TimeoutException: null
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:65)
	... 49 common frames omitted
```
